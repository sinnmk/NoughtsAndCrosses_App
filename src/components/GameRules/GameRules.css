.game-rules-header{
    border-block-end-style: dashed; 
}
.game-body-container{
    background-color: cadetblue;
}