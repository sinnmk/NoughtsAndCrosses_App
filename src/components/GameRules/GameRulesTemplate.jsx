import React from 'react';
import {Link} from 'react-router-dom';
import {Button} from 'reactstrap';
import Home from '../Home/Home';

const GameRulesTemplate = () =>{
    return( 
        <div>
            <Link to="/" onClick={Home}><Button className = 'back-button'>Back to Main Menu</Button></Link>
            <header className="card-header">
                <h4 className= 'card-header--title'>Game Rules</h4>
            </header>
            <div className="card-body">
                <p className="body-content">
                    The object of Tic Tac Toe is to get three in a row. 
                    You play on a three by three game board. 
                    The first player is known as X and the second is O. 
                    Players alternate placing Xs and Os on the game board 
                    until either oppent has three in a row or all nine squares are filled.
                </p>
            </div>
        </div>
    )
}
export default GameRulesTemplate;