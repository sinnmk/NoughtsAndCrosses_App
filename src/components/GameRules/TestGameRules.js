import React from 'react';
import {shallow, mount, render} from 'enzyme';
import GameRules from "./GameRules";

describe("GameRules", () => {
    it("can render the GameRules component", () => {

    });
});