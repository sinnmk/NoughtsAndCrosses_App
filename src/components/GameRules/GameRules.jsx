import {Component} from 'react';
import GameRulesTemplate from './GameRulesTemplate';

export default class GameRules extends Component{

    constructor(props){
        super(props);
        this.state = {}
    }

    render(){
        return GameRulesTemplate.call(this);
    }
    
    backToMenu(){

    }
}
