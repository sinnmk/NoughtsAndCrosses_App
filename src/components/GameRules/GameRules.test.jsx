import React from 'react';
import renderer from 'react-test-renderer';
import GameRules from './GameRules';

describe('When GameRules is Clicked', () =>{
    test('Game rules content is rendered', () =>{
        const component = renderer.create(
            <GameRules/>).toJSON();

        expect(component).toMatchSnapshot();
    })
})