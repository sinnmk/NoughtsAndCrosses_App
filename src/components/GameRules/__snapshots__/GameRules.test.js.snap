// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When GameRules is Clicked Game rules content is rendered 1`] = `
<div>
  <header
    className="card-header"
  >
    <h4
      className="card-header--title"
    >
      Game Rules
    </h4>
  </header>
  <div
    className="card-body"
  >
    <p
      className="body-content"
    >
      The object of Tic Tac Toe is to get three in a row. You play on a three by three game board. The first player is known as X and the second is O. Players alternate placing Xs and Os on the game board until either oppent has three in a row or all nine squares are filled.
    </p>
  </div>
</div>
`;
